java/pom.xml

```diff
@@ -905,6 +905,197 @@
       </build>
     </profile>
 
+    <profile>
+      <id>generate-dataset-dylib_so</id>
+      <properties>
+        <arrow.dataset.jni.dist.dir>java-dist/lib</arrow.dataset.jni.dist.dir>
+        <cpp.dependencies.builded>false</cpp.dependencies.builded>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <version>3.1.0</version>
+            <executions>
+              <execution>
+                <id>datasetdir</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>mkdir</executable>
+                  <commandlineArgs>-p cpp-jni java-dist java-jni</commandlineArgs>
+                  <workingDirectory>../</workingDirectory>
+                </configuration>
+              </execution>
+              <execution>
+                <id>datasetarrowdependency</id>
+                <phase>generate-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>cmake</executable>
+                  <commandlineArgs>
+                    -S cpp
+                    -B cpp-jni
+                    -DARROW_DEPENDENCY_USE_SHARED=OFF
+                    -DARROW_CSV=ON
+                    -DARROW_DATASET=ON
+                    -DARROW_FILESYSTEM=ON
+                    -DARROW_GANDIVA=ON
+                    -DARROW_GANDIVA_JAVA=ON
+                    -DARROW_GANDIVA_STATIC_LIBSTDCPP=ON
+                    -DARROW_JNI=ON
+                    -DARROW_ORC=ON
+                    -DARROW_PARQUET=ON
+                    -DARROW_PLASMA=ON
+                    -DARROW_PLASMA_JAVA_CLIENT=ON
+                    -DARROW_S3=ON
+                    -DARROW_USE_CCACHE=ON
+                    -DCMAKE_BUILD_TYPE=Release
+                    -DCMAKE_INSTALL_LIBDIR=lib
+                    -DCMAKE_INSTALL_PREFIX=java-dist
+                    -DCMAKE_UNITY_BUILD=ON
+                    -DARROW_DEPENDENCY_SOURCE=BUNDLED
```

Comment:
Could you sort this list in alphabetical order?

Reply:
Sorted